bottlecap/src/config/mod.rs

```diff
@@ -177,17 +177,26 @@ fn fallback(figment: &Figment, yaml_figment: &Figment) -> Result<(), ConfigError
         };
 
     // Customer explicitly opted out of the Next Gen extension
-    let opted_out = match config.extension_version.as_deref() {
+    let is_compatibility = match config.extension_version.as_deref() {
         Some("compatibility") => true,
         // We want customers using the `next` to not be affected
         _ => false,
     };
 
+    // Check for us-gov region
+    let opted_out = is_compatibility || region.starts_with("us-gov-");
+
     if opted_out {
-        log_fallback_reason("extension_version");
-        return Err(ConfigError::UnsupportedField(
-            "extension_version".to_string(),
-        ));
+        log_fallback_reason(if is_compatibility {
+            "extension_version"
+        } else {
+            "gov_region"
+        });
+        return Err(ConfigError::UnsupportedField(if is_compatibility {
+            "extension_version".to_string()
+        } else {
+            "gov_region".to_string()
+        }));
```

Comment:
govcloud should be its own section, just like we do below for appsec or otel, update this code to reflect that